runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -1945,7 +1949,101 @@ void TR_EscapeAnalysis::checkDefsAndUses()
       }
    }
 
+void printUsesThroughAternary(TR::Compilation *comp, TR_HashTab *usesTable)
+   {
+   traceMsg(comp, "\nNodes used through aternary operations\n");
+   TR_HashTabIterator hti(usesTable);
+   TR_HashId nextID;
+   TR_Array<TR::Node*> *aternaryUses = (TR_Array<TR::Node*>*) hti.getFirst(nextID);
+
+   while (aternaryUses)
+      {
+      TR::Node *key = (TR::Node*) usesTable->getKey(nextID);
+      int32_t nodeIdx = key->getGlobalIndex();
+
+      traceMsg(comp, "   node [%p] n%dn is used by {", key, nodeIdx, aternaryUses);
+
+      TR_ArrayIterator<TR::Node> ai(aternaryUses);
+      bool first = true;
+
+      for (TR::Node* aternaryNode = ai.getFirst(); aternaryNode; aternaryNode = ai.getNext())
+         {
+         traceMsg(comp, "%s[%p] n%dn", (first ? "" : ", "), aternaryNode,
+                  aternaryNode->getGlobalIndex());
+         first = false;
+         }
+
+      traceMsg(comp, "}\n");
+
+      aternaryUses = (TR_Array<TR::Node*>*) hti.getNext(nextID);
+      }
+   }
+
+void TR_EscapeAnalysis::gatherUsesThroughAternary()
+   {
+   TR::NodeChecklist visited(comp());
+   TR::TreeTop *tt = comp()->getStartTree();
 
+   for (; tt; tt = tt->getNextTreeTop())
+      {
+      TR::Node *node = tt->getNode();
+      gatherUsesThroughAternary(node, visited);
+      }
+
+   if (trace() && _nodeUsesThroughAternary)
+      {
+      printUsesThroughAternary(comp(), _nodeUsesThroughAternary);
+      }
+   }
+
+void TR_EscapeAnalysis::gatherUsesThroughAternary(TR::Node *node, TR::NodeChecklist& visited)
+   {
+   if (visited.contains(node))
+      {
+      return;
+      }
+   visited.add(node);
+
+   if (_nodeUsesThroughAternary == NULL
+          && node->getOpCode().isTernary()
+          && node->getDataType() == TR::Address)
+      {
+      _nodeUsesThroughAternary = new (trStackMemory()) TR_HashTab(trMemory(), stackAlloc);
+      }
+
+   int32_t i;
+   for (i=0;i<node->getNumChildren(); i++)
```

Comment:
why the larger scope for I than the loop?

Reply:
would this make more sense unrolled so as to process each child explicitly since ternaries have a fixed number of children - the loop makes this seem more general than it needs to be...

Reply:
> why the larger scope for I than the loop?

Hmmm.  I might have started off planning to break out of the loop early or something.  Thanks for catching that!

> would this make more sense unrolled so as to process each child explicitly since ternaries have a fixed number of children

The recursive `gatherUsesThroughAternary` applies more generally, so that call needs to stay in a loop over `[0..getNumChildren())`.  But I will pull out the portion dealing with the second and third operands of `aternary`.

Reply:
Andrew @andrewcraik, both of these are fixed in https://github.com/eclipse/openj9/pull/5699/commits/d10e03a13121fc002836f50c90327408cdfd4f63